Dfe.Academies.External.Web/Pages/School/Leases.cshtml.cs

```diff
@@ -54,7 +51,7 @@ public bool HasError
 			}
 		}
 
-		public async Task<IActionResult> OnPostAsync()
+		public override async Task<IActionResult> OnPostAsync()
 		{
 			var selectedSchool = await LoadAndSetSchoolDetails(ApplicationId, Urn);
 			MergeCachedAndDatabaseEntities(selectedSchool);
```

Comment:
in the OnPost, can you replace the two if's with a call to this :- RunUiValidation()